fuzz.go

```diff
@@ -61,6 +63,33 @@ func NewWithSeed(seed int64) *Fuzzer {
 	return f
 }
 
+// NewFromGoFuzz is a helper function that enables using gofuzz (this
+// project) with go-fuzz (https://github.com/dvyukov/go-fuzz) for continuous
+// fuzzing. Essentially, it enables translating the fuzzing bytes from
+// go-fuzz to any Go object using this library.
+//
+// Example: use go-fuzz to test the function `MyFunc(int)` in the package
+// `mypackage`. Add the file: "mypacakge_fuzz.go" with the content:
+//
+// // +build gofuzz
+// package mypacakge
+// import "github.com/google/go-fuzz"
+// func Fuzz(data []byte) int {
+// 	var i int
+// 	fuzz.NewFromGoFuzz(data).Fuzz(&i)
+// 	MyFunc(i)
+// 	return 0
+// }
+func NewFromGoFuzz(data []byte) *Fuzzer {
+	maxint := big.NewInt(math.MaxInt64)
+	seed := big.NewInt(1)
+	for _, b := range data {
+		seed.Mul(seed, big.NewInt(int64(b)))
+		seed.Mod(seed, maxint)
+	}
+	return NewWithSeed(seed.Int64())
```

Comment:
Let me suggest an alternative:
* We'll make a type that implements rand.Source and rand.Source64, construct a rand.Rand with that source, then construct a *Fuzzer with that rand.Rand.
* We want the property that the caller can effectively search the whole space; we don't want it to be the case that every input randomizes everything.
* So I think we want to divide the data into two pieces:
  * bytes we can directly consume and return from the rand.Source
  * A seed we'll use for generating everything else when we run out of bytes

Make sense? I think this design will let https://github.com/dvyukov/go-fuzz actually gradually find more interesting objects over time faster than making every seed do something completely different.

Reply:
I see your concern, but can't fully understand the API that you propose.
Do you suggest to store part of `data` in the `Fuzzer` object, and let the user consume it as it would have done with just using `data` directly?

Reply:
Right, just to illustrate the idea more

```
type guidedFuzzingSource struct {
  input []byte

  fallback rand.Source
}

func (g *guidedFuzzingSource) Uint64() uint64 {
  if len(g.input) > 8 {
    // consume first 8 bytes: shorten g.input by 8, convert those bytes to uint64, return
  } else if len(g.input) > 0 {
    // consum remaining bytes, construct rand.Source, store it as g.fallback
  }
  return g.fallback.Uint64() 
}

...

r := rand.New(&guidedFuzzingSource{input: data})

```

Reply:
Oh, got it, this is great stuff.